java/maven/src/org/netbeans/modules/maven/ActionProviderImpl.java

```diff
@@ -570,6 +570,7 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
             Object retValue = DialogDisplayer.getDefault().notify(dd);
             if (retValue == DialogDescriptor.OK_OPTION) {
                 pnl.applyValues(mapping);
+                maps.getActions().remove(mapping); // deduplicate
```

Comment:
Is this the main fix?

Reply:
no, this is just convenience that the same item isn't repeatedly stored in history if you re-run it. marked the main fix via comment